src/utils/sourcemaps.rs

```diff
@@ -635,8 +669,71 @@ impl SourceMapProcessor {
                 continue;
             }
 
-            if let Some(Some(sourcemap_url)) = self.sourcemap_references.get(&source.url) {
-                source.set_sourcemap_reference(sourcemap_url.to_string());
+            if let Some(Some(sourcemap)) = self.sourcemap_references.get(&source.url) {
+                source.set_sourcemap_reference(sourcemap.url.to_string());
+            }
+        }
+        Ok(())
+    }
+
+    /// Adds debug id reference from the given source map to all minified files.
+    /// This is used for files we can't read debug ids from (e.g. Hermes bytecode bundles).
+    pub fn add_debug_id_references(&mut self) -> Result<()> {
+        self.flush_pending_sources();
+
+        for source in self.sources.values_mut() {
+            if source.ty != SourceFileType::MinifiedSource {
+                continue;
+            }
+
+            if let Some(Some(sourcemap_reference)) = self.sourcemap_references.get(&source.url) {
+                let sourcemap_url = &sourcemap_reference
+                    .original_url
+                    .clone()
+                    .unwrap_or(sourcemap_reference.url.clone());
+
+                if !self.debug_ids.contains_key(sourcemap_url) {
+                    debug!(
+                        "{} No debug id found for {} to reference",
+                        style(">").dim(),
+                        sourcemap_url
+                    );
+                    return Ok(());
+                }
+
+                let debug_id_header = (
+                    "debug-id".to_string(),
+                    self.debug_ids[sourcemap_url].to_string(),
+                );
+                if source.headers.contains_key(&debug_id_header.0) {
```

Comment:
I added methods `debug_id` and `set_debug_id` to the `SourceFile` struct recently that let you do this without worrying about the headers.

Reply:
Ohh, nice. I didn't know.